### ISSUE TITLE
Issues detecting Boost on OpenSuse.

### ISSUE THREAD

**Issue by Zombie-Ryushu** [cmangos/wotlk-db#395]
I am trying to compile for Boost 1.66 on OpenSuse 15.3, and I keep getting missing dependancy errors.
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Boost (missing: system program_options thread regex
  serialization filesystem) (found version "1.66.0")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindBoost.cmake:2145 (find_package_handle_standard_args)
  CMakeLists.txt:215 (find_package)

**Comment by berserkingyadis** [comment]
Please use the bug template and the right repo to report issues. Also please provide your cmake version.

**Comment by berserkingyadis** [comment]
https://github.com/cmangos/issues/issues/new/choose

**Comment by AdamMajer** [comment]
The problem that @Zombie-Ryushu has found is that distros, in general, don't ship static libraries and your CMakeLists.txt only searches for static libraries.

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e7086d58c..91881cc63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -191,10 +191,6 @@ if(WIN32 AND NOT "$ENV{BOOST_ROOT}" STREQUAL "")
   set(BOOST_LIBRARYDIR "${BOOST_LIBRARYDIR};$ENV{BOOST_ROOT}\\stage")
 endif()
 
-# Be sure to search only static lib
-add_definitions(-DBOOST_ALL_NO_LIB)
-set(Boost_USE_STATIC_LIBS ON)
-set(Boost_NO_BOOST_CMAKE ON)   # should fix build for boost 1.71
 #set(Boost_DEBUG ON)
 
 if(MSVC)
```

fixed the issue and everything appears to compile and link correctly.

**Comment by berserkingyadis** [comment]
Thank you Adam for providing a solution here. :)
